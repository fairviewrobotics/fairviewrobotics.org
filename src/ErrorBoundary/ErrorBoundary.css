.error-container {
  display: flex;
  min-height: 100%;
  flex-grow: 1;
}

.error-item {
  margin: auto;
  text-align: center;
}