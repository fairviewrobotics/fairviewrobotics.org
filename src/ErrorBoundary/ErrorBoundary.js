import React, { Component } from 'react';
import styles from './ErrorBoundary.module.css';

export default class ErrorBoundary extends Component {

  state = {
    hasError: false
  };

  componentDidCatch() {
    this.setState({ hasError: true })
  }

  render() {

    if (this.state.hasError) {
      return (
        <div className={styles.container}>
          <div className={styles.errorMessage}>
            <h1>Something went wrong!</h1>
            <p>Try refreshing the browser.</p>
          </div>
        </div>
      );
    }

    return this.props.children;
  }
}
