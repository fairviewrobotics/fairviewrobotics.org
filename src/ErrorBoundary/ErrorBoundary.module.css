.container {
  display: flex;
  min-height: 100%;
}

.errorMessage {
  margin: auto;
  text-align: center;
}