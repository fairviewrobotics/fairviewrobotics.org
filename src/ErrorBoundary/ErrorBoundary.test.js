import React, { Component } from "react";
import { mount, shallow } from "enzyme";
import ErrorBoundary from "./ErrorBoundary";

const setup = props => {
  const defaultProps = {};
  const finalProps = {...defaultProps, ...props};

  const wrapper = shallow(
    <ErrorBoundary {...finalProps} />
  );
  return { wrapper: wrapper }
};

describe('ErrorBoundary', () => {
  let wrapper;

  it('renders without crashing', () => {
    setup();
  });

  describe('when a child component crashes', () => {
    beforeEach(() => {

      class ThrowError extends Component {
        render() {
          throw new Error();
        }
      }

      wrapper = mount(
        <ErrorBoundary>
          <ThrowError />
        </ErrorBoundary>
      );
    });

    it('renders a error message to the user', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when the children work as intended', () => {
    beforeEach(() => {
      wrapper = shallow(
        <ErrorBoundary/>
      )
    });

    it('renders the children', () => {
      const { wrapper } = setup({ children: (
        <div>
          <h1>Page Name</h1>
          <p>Robots are cool!</p>
        </div>
        ) });

      expect(wrapper).toMatchSnapshot();
    });
  });
});