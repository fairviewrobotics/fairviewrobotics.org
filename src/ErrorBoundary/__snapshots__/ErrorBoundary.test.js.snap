// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary when a child component crashes renders a error message to the user 1`] = `
<div
  className="container"
>
  <div
    className="errorMessage"
  >
    <h1>
      Something went wrong!
    </h1>
    <p>
      Try refreshing the browser.
    </p>
  </div>
</div>
`;

exports[`ErrorBoundary when the children work as intended renders the children 1`] = `
<div>
  <h1>
    Page Name
  </h1>
  <p>
    Robots are cool!
  </p>
</div>
`;
